ivy/functional/frontends/paddle/tensor/tensor.py

```diff
@@ -240,3 +244,10 @@ def astype(self, dtype):
     )
     def logical_xor(self, y, out=None, name=None):
         return paddle_frontend.logical_xor(self, y, out=out)
+
+    @with_supported_device_and_dtypes(
+        {"2.5.0 and below": {"cpu": ("float32", "float64", "complex128", "complex64")}},
+        "paddle",
+    )
+    def eig(self):
+        return ivy.eig(self._ivy_array)

```

Comment:
There is already an implementation of `eig` in the `paddle` frontend. Kindly use that here

Reply:
Maybe i'm getting this wrong, but when calling the ivy.eig() i'm using the eig function defined in the linear_algebra file, I think you might be referring to that. Is this not what you mean?

Reply:
What @hmahmood24 is referring to is use something like `paddle_frontend.eig(self)` as this is an instance method